### PR TITLE
docs(component-styles): drop native view encapsulation

### LINKED ISSUE #1365: Remove docs for native view encapsulation
According to dart-lang/angular#997, we haven't supported native view encapsulation since v4.

### LINKED ISSUE #1376: Special CSS selectors should only be used in emulated encapsulation mode
From @leonsenft in https://github.com/dart-lang/site-webdev/pull/1373#issuecomment-370519931:

> Awesome, thanks @chalin. I noticed there's a warning unique to `::ng-deep` indicating it in particular should only be used with emulated encapsulation, but this warning should really apply to all three special selectors now. Perhaps we could move this to the beginning of the section?
> 
> https://angulardart-org-dev2.firebaseapp.com/angular/guide/component-styles#special-selectors

### PR DESCRIPTION
**Pull request by chalin**
Fixes #1365, https://github.com/dart-lang/angular/issues/997
Fixes #1376

cc @leonsenft 

Staged at https://angulardart-org-dev2.firebaseapp.com/angular/guide/component-styles#controlling-view-encapsulation

### CODE REVIEW

**Comment by chalin** [comment]
@leonsenft - FYI, after this PR is approved and merged, the 5-dev branch will be rebased over it.

**Comment by leonsenft** [comment]
Awesome, thanks @chalin. I noticed there's a warning unique to `::ng-deep` indicating it in particular should only be used with emulated encapsulation, but this warning should really apply to all three special selectors now. Perhaps we could move this to the beginning of the section?

https://angulardart-org-dev2.firebaseapp.com/angular/guide/component-styles#special-selectors

**Comment by chalin** [comment]
Thanks for noticing that @leonsenft. I've created #1376 to track that fix to the docs.

File: src/angular/guide/component-styles.md [hunk 1]
```diff
@@ -267,44 +267,32 @@ In *this* case the URL is relative to the CSS file into which we are importing.
...
-  on the [MDN](https://developer.mozilla.org) site)
-  to attach a shadow DOM to the component's host element, and then puts the component
-  view inside that shadow DOM. The component's styles are included within the shadow DOM.
-* `Emulated` view encapsulation (the default) emulates the behavior of shadow DOM by preprocessing
-  (and renaming) the CSS code to effectively scope the CSS to the component's view.
+<a id="view-encapsulation"></a>
+## Controlling view encapsulation
+
+As discussed earlier, component CSS styles are encapsulated within a component's view,
+so they don't affect the rest of the app.
```
Comment (kwalrath) [inline src/angular/guide/component-styles.md]: The "so" is a bit unclear, and now that there are only two values (the default and none), I think we can be cleaner. Maybe:

---------

By default, component styles affect only the component, not the rest of the app. To change this behavior, you can either use the [::ng-deep selector][] or completely remove the component's view encapsulation.

Removing view encapsulation makes the component's styles global.
To do this, set the component's metadata [encapsulation][] parameter to
`ViewEncapsulation.None`:

```
...code here...
```

The ViewEncapsulation enum can have two values:

* Emulated...
* None...

Comment (chalin) [reply src/angular/guide/component-styles.md]: I've reworked the text along the lines that you have suggested. PTAL.

**Comment by chalin** [comment]
> I noticed there's a warning unique to ::ng-deep indicating it in particular should only be used with emulated encapsulation, but this warning should really apply to all three special selectors now. Perhaps we could move this to the beginning of the section?

@leonsenft: done, see https://angulardart-org-dev2.firebaseapp.com/angular/guide/component-styles#special-selectors

File: src/angular/guide/component-styles.md [hunk 2]
```diff
@@ -270,19 +269,14 @@ In *this* case the URL is relative to the CSS file into which we are importing.
...
 
-As discussed earlier, component CSS styles are encapsulated within a component's view,
-so they don't affect the rest of the app.
-
-To control whether encapsulation happens on a _per component_ basis,
-set the component's metadata [encapsulation][] parameter
-to one of [ViewEncapsulation][] enum values:
+By default, component styles are _encapsulated_, affecting only the HTML in the
+component's template. You can use the [special selectors](#special-selectors)
+to reach outside of the component's view, or disable view encapsulation
```
Comment (kwalrath) [inline src/angular/guide/component-styles.md]: , or disable -> , or you can disable

(It was unclear whether the special selectors disabled view encapsulation.)

Comment (chalin) [reply src/angular/guide/component-styles.md]: Right, thanks for catching that. Fixed.

**Pull request merged**